staging/src/k8s.io/apiserver/pkg/admission/interfaces.go

```diff
@@ -61,6 +61,16 @@ type Interface interface {
 	Handles(operation Operation) bool
 }
 
+// ValidationInterface is an abstract, pluggable interface for Admission Control decisions.
+type ValidationInterface interface {
+	// Validate makes an admission decision based on the request attributes.  It is NOT allowed to mutate
+	Validate(a Attributes) (err error)
+
+	// Handles returns true if this admission controller can handle the given operation
+	// where operation can be one of CREATE, UPDATE, DELETE, or CONNECT
+	Handles(operation Operation) bool
```

Comment:
If this is called like in the mutating `Interface`, we cannot provide different implementations for mutation and validations in the same object.